src/sentry/relay/config/metric_extraction.py

```diff
@@ -509,26 +489,35 @@ def _convert_aggregate_and_query_to_metric(
         if not should_use_on_demand_metrics(dataset, aggregate, query, groupbys, prefilling):
             return None
 
-        on_demand_spec = OnDemandMetricSpec(
-            field=aggregate,
-            query=query,
-            environment=environment,
-            groupbys=groupbys,
-            spec_type=spec_type,
-            use_updated_env_logic=use_updated_env_logic,
-        )
-
-        metric_spec = on_demand_spec.to_metric_spec(project)
-        # TODO: switch to validate_rule_condition
-        if (condition := metric_spec.get("condition")) is not None:
-            validate_sampling_condition(json.dumps(condition))
-        else:
-            metrics.incr(
-                "on_demand_metrics.missing_condition_spec",
-                tags={"prefilling": prefilling},
+        metric_specs_and_hashes = []
+        hashes = set()
+        # Create as many specs as we support without duplication
+        for spec_version in SPEC_VERSIONS:
+            on_demand_spec = OnDemandMetricSpec(
+                field=aggregate,
+                query=query,
+                environment=environment,
+                groupbys=groupbys,
+                spec_type=spec_type,
+                spec_version=spec_version,
             )
+            metric_spec = on_demand_spec.to_metric_spec(project)
+            # This prevents returning duplicates
+            if on_demand_spec.query_hash in hashes:
```

Comment:
Why are we running deduplication here? We already do it upstream when merging specs.

Reply:
I will investigate getting rid of it this afternoon.